server/neptune/workflows/internal/deploy/revision/revision.go

```diff
@@ -0,0 +1,41 @@
+package revision
+
+import (
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/revision/queue"
+	"go.temporal.io/sdk/workflow"
+)
+
+type NewRevisionRequest struct {
+	Revision string
+}
+
+type Queue interface {
+	Push(queue.Message)
+}
+
+func NewReceiver(ctx workflow.Context, queue Queue) *Receiver {
+	return &Receiver{
+		queue: queue,
+		ctx:   ctx,
```

Comment:
Per https://pkg.go.dev/context
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. 

Is that an option here? If not, why not?

Reply:
Ah looking at the call site it's because `Selector.AddReceive` wants a `func(workflow.ReceiveChannel, bool)`. But since we're already creating a closure to pass in, WDYT about passing in the context we have there, so that this has access to the context without needing to store it?